mathics/builtin/linalg.py

```diff
@@ -243,6 +244,26 @@ def apply(self, m, evaluation):
         return Expression('List', *[U_list, S_list, V_list])
 
 
+class QRDecomposition(Builtin):
+    """
+    <dl>
+    <dt>'QRDecomposition[$m$]'
+        <dd>computes the QR decomposition of the matrix $m$.
+    </dl>
+
+    >> QRDecomposition[{{1, 2}, {3, 4}, {5, 6}}]
+     = {{{Sqrt[35] / 35, 3 Sqrt[35] / 35, Sqrt[35] / 7}, {13 Sqrt[210] / 210, 2 Sqrt[210] / 105, -Sqrt[210] / 42}}, {{Sqrt[35], 44 Sqrt[35] / 35}, {0, 2 Sqrt[210] / 35}}}
+    """
+    
+    def apply(self, m, evaluation):
+        'QRDecomposition[m_]'
```

Comment:
What happens if we give something that's not a matrix? e.g. QRDecomposition[4x + y]